File: backend/src/main/java/com/gfi/parkplatzapp/backend/persistence/entities/Parkflaeche.java
package com.gfi.parkplatzapp.backend.persistence.entities;

import com.gfi.parkplatzapp.backend.utils.AktivitaetEnum;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class Parkflaeche {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long parkflaecheID;

    private String bezeichnung;

    @Enumerated(EnumType.STRING)
    private AktivitaetEnum aktivitaet;

    @ManyToOne
    private DBImage image;

    @ManyToOne
    private Parkhaus parkhaus;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Parkplatz> parkplatzList;

}

File: frontend/src/app/facade/Parkflaeche.ts
import { AktivitaetsEnum } from "../utils/aktivitaetEnum.utils"
import { Parkplatz } from "./Parkplatz"

export interface Parkflaeche {
    parkflaecheID: number
    bezeichnung: string
    aktivitaet: AktivitaetsEnum
    parkplatzList: Parkplatz[]
}